### PR TITLE
AWSGems: Update all references to CDKv2

### PR DESCRIPTION
**Pull request by lmbr-pip**
## Change summary

AWSGems are migrating to CDKv2 in this PR: https://github.com/o3de/o3de/pull/11675

This is the related docs change to reflect this.

Also fixes:
* Some self-referencing mistakes when trying to provide instructions about enabling gems.
* Minor spelling, language or other issues noticed during cleanup.

### Submission Checklist:

* [x] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [x] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [x] **Consistency** - Does the content consistently follow the [style guide](https://o3de.org/docs/contributing/to-docs/style-guide/)?
* [x] **Help the user** - Does the documentation show the user something *meaningful*?

**Note**: Should only be merged once https://github.com/o3de/o3de/pull/11675 has been merged.

### CODE REVIEW

**Review by junbo75 — APPROVED** [review]

File: content/docs/user-guide/gems/reference/aws/aws-core/cdk-application.md
Comment (FiniteStateGit) [inline content/docs/user-guide/gems/reference/aws/aws-core/cdk-application.md:108]:
```suggestion
For more information about the bootstrap provisioning process, see the [Bootstrapping](https://docs.aws.amazon.com/cdk/v2/guide/bootstrapping.html) section of the AWS CDK Developer Guide.
```

File: content/docs/user-guide/gems/reference/aws/aws-core/configuring-credentials.md
Comment (FiniteStateGit) [inline content/docs/user-guide/gems/reference/aws/aws-core/configuring-credentials.md:182]:
```suggestion
You can attach IAM policies to a user group to control the permissions they have access to. This can be done using either the AWS Console, the AWS CLI, CloudFormation template or with the CDK. See [Attaching a policy to an IAM user group ](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_groups_manage_attach-policy.html) to attach permissions using the AWS Console or AWS CLI.
```
Nit: We've been avoiding Latinisms.

File: content/docs/user-guide/gems/reference/aws/aws-metrics/setup.md
Comment (FiniteStateGit) [inline content/docs/user-guide/gems/reference/aws/aws-metrics/setup.md:31]:
```suggestion
If you haven't already added and built the **AWS Metrics Gem** in your project, follow the steps to [add the Gem](/docs/user-guide/project-config/add-remove-gems/) to your current project.
```

File: content/docs/user-guide/gems/reference/aws/aws-core/cdk-application.md
Comment (FiniteStateGit) [inline content/docs/user-guide/gems/reference/aws/aws-core/cdk-application.md:19]: There is no issue with renumbering this, but Hugo will automatically number lists that only use `1.` for each step.

**Review by FiniteStateGit — CHANGES_REQUESTED** [review]
@lmbr-pip You should amend your DCO signoff before applying these suggestions.

File: content/docs/user-guide/gems/reference/aws/aws-core/cdk-application.md
Comment (lmbr-pip) [reply content/docs/user-guide/gems/reference/aws/aws-core/cdk-application.md:19]: This was part of an automated cleanup, confirmed they do display directly but will try to avoid renumbering as it prevents future breakage when folks add new items.

File: content/docs/user-guide/gems/reference/aws/aws-client-auth/authentication-providers.md
Comment (allisaurus) [inline content/docs/user-guide/gems/reference/aws/aws-client-auth/authentication-providers.md:31]: prefer full [official](https://aws.amazon.com/cdk/) name: "AWS **Cloud** Development Kit (AWS CDK)"

File: content/docs/user-guide/gems/reference/aws/aws-core/cdk-application.md
Comment (allisaurus) [inline content/docs/user-guide/gems/reference/aws/aws-core/cdk-application.md:8]: similar to above, prefer full official name and acronym ("AWS Cloud..." / "(AWS CDK)")

File: content/docs/user-guide/gems/reference/aws/aws-core/configuring-credentials.md
Comment (allisaurus) [inline content/docs/user-guide/gems/reference/aws/aws-core/configuring-credentials.md:147]: "~~Cloudformation~~..." --> AWS Cloud Development Kit (AWS CDK)

File: content/docs/user-guide/gems/reference/aws/aws-gamelift/resource-management.md
Comment (allisaurus) [inline content/docs/user-guide/gems/reference/aws/aws-gamelift/resource-management.md:9]: "~~Cloudformation~~..." --> AWS Cloud Development Kit (AWS CDK)

File: content/docs/user-guide/gems/reference/aws/aws-gamelift/resource-management.md
Comment (allisaurus) [inline content/docs/user-guide/gems/reference/aws/aws-gamelift/resource-management.md:20]: "~~Cloudformation~~..." --> AWS Cloud Development Kit (AWS CDK)

File: content/docs/user-guide/gems/reference/aws/aws-metrics/using.md
Comment (allisaurus) [inline content/docs/user-guide/gems/reference/aws/aws-metrics/using.md:12]: "~~Cloudformation~~..." --> AWS Cloud Development Kit (AWS CDK)

**Review by allisaurus — CHANGES_REQUESTED** [review]

**Comment by willihay** [comment]
@lmbr-pip Do you expect the linked PR in the source repo to be pulled into stabilization? If so, we'll want to either retarget this PR to the docs stabilization/22.10, or cherry pick later.

**Comment by lmbr-pip** [comment]
No it should not go to stabilization, CDKv2 changes are not at this time considered in scope for 2210.

**Review by FiniteStateGit — APPROVED** [review]

File: content/docs/user-guide/gems/reference/aws/aws-client-auth/authentication-providers.md
Comment (micronAMZN) [inline content/docs/user-guide/gems/reference/aws/aws-client-auth/authentication-providers.md:31]:
```suggestion
When deploying the optional AWS Cloud Development Kit (AWS CDK) application, be sure to use the CDK constant that corresponds to your selected provider. Refer to the CDK application deployment step in [Setting Up Client Auth](./setup/) for details.
```

Comment (lmbr-pip) [reply content/docs/user-guide/gems/reference/aws/aws-client-auth/authentication-providers.md:31]: Fixed.

File: content/docs/user-guide/gems/reference/aws/aws-core/cdk-application.md
Comment (lmbr-pip) [reply content/docs/user-guide/gems/reference/aws/aws-core/cdk-application.md:8]: Fixed

**Review by micronAMZN — APPROVED** [review]
Approved pending updates to **AWS Cloud Development Kit (AWS CDK)** references.

File: content/docs/user-guide/gems/reference/aws/aws-gamelift/resource-management.md
Comment (lmbr-pip) [reply content/docs/user-guide/gems/reference/aws/aws-gamelift/resource-management.md]: Fixed

File: content/docs/user-guide/gems/reference/aws/aws-gamelift/resource-management.md
Comment (lmbr-pip) [reply content/docs/user-guide/gems/reference/aws/aws-gamelift/resource-management.md]: Fixed

File: content/docs/user-guide/gems/reference/aws/aws-metrics/using.md
Comment (lmbr-pip) [reply content/docs/user-guide/gems/reference/aws/aws-metrics/using.md]: Fixed

File: content/docs/user-guide/gems/reference/aws/aws-client-auth/authentication-providers.md
Comment (lmbr-pip) [reply content/docs/user-guide/gems/reference/aws/aws-client-auth/authentication-providers.md]: I fixed all these I believe.

File: content/docs/user-guide/gems/reference/aws/aws-core/cdk-application.md
Comment (willihay) [inline content/docs/user-guide/gems/reference/aws/aws-core/cdk-application.md:13]:
```suggestion
For those still using the AWS CDK v1, a legacy snapshot of the sample applications are available in a `cdkv1` folder inside each Gem. See the [AWS CDK Migration Guide](https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html) for advice about how to migrate legacy AWS CDK applications to the AWS CDK v2. 
```

File: content/docs/user-guide/gems/reference/aws/aws-core/cdk-application.md
Comment (willihay) [inline content/docs/user-guide/gems/reference/aws/aws-core/cdk-application.md:137]:
```suggestion
| `cdk docs` | Open AWS CDK documentation.                           |
```

**Review by willihay — CHANGES_REQUESTED** [review]

**Review by allisaurus — APPROVED** [review]

**Comment by lmbr-pip** [comment]
Due to potential issues jumping back and forth from CDKv1 to CKDv2, guide needs a small addition to cover troubleshooting this.

Also will be holding this and docs changes pending 2210 release to avoid any destabilization on Jenkins until release has cleared due to CDK version bump.

**Review by willihay — APPROVED** [review]

**Comment by willihay** [comment]
Adding a do-not-merge hold per author.

**Comment by lmbr-pip** [comment]
Team will refresh this PR and move to final form in the week of 10/31. Will address https://github.com/o3de/o3de/issues/12848

**Pull request merged**
